### PR TITLE
Backoff message collector

### PR DESCRIPTION
**Pull request by annafossen**

### CODE REVIEW

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java [hunk 1]
```diff
@@ -118,33 +118,33 @@ public void shouldCheckSubscriptionRequestsForUpdates_neighbourWithNewCapability
 		Neighbour neighbourWithNewerCapabilitiesThanSubscriptionRequest = new Neighbour("nice-neighbour", capabilities, null, fedIn);
 		assertThat(neighbourWithNewerCapabilitiesThanSubscriptionRequest.shouldCheckSubscriptionRequestsForUpdates(Optional.of(localSubscriptionUpdateTimeBeforeCapabilityPost))).isTrue();
 	}
-
+	//Flytte til en egen test-klasse som tester ConnectionBackoff
```
Comment (kyrrek) [inline neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java]: REmove this test if it's already in ConnectionBackupTest

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java [hunk 2]
```diff
@@ -119,34 +119,6 @@ public void shouldCheckSubscriptionRequestsForUpdates_neighbourWithNewCapability
...
-		assertThat(neighbour.getConnectionStatus()).isEqualTo(ConnectionStatus.UNREACHABLE);
-
-		neighbour.okConnection();
-		assertThat(neighbour.getConnectionStatus()).isEqualTo(ConnectionStatus.CONNECTED);
-		assertThat(neighbour.getBackoffStartTime()).isNull();
-		assertThat(neighbour.getBackoffAttempts()).isEqualTo(0);
-	}
-
 	@Test
 	public void calculatedNextPostAttemptTimeIsInCorrectInterval(){
```
Comment (kyrrek) [inline neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java]: Take out the reference to Neighbour, move the test to ConnectionBackoffTest, working directly on the ConnectionBakoff class

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java (hunk 1 above)
Comment (annafossen) [reply neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java]: Done :)

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java (hunk 2 above)
Comment (annafossen) [reply neighbour-model/src/test/java/no/vegvesen/ixn/federation/model/NeighbourTest.java]: Done :)

File: message-collector/src/main/java/no/vegvesen/ixn/federation/messagecollector/MessageCollector.java [hunk 3]
```diff
@@ -52,39 +57,51 @@ private void checkListenerList() {
...
         }
+
         List<String> listenerKeysToRemove = new ArrayList<>();
         for (String ixnName : listeners.keySet()) {
-            if (! interchangeNames.contains(ixnName)) {
+            if (!interchangeNames.contains(ixnName)) {
                 logger.info("Listener for {} is now being removed",ixnName);
-
                 MessageCollectorListener toRemove = listeners.get(ixnName);
-                logger.info("Tearing down {}", ixnName);
```
Comment (kyrrek) [inline message-collector/src/main/java/no/vegvesen/ixn/federation/messagecollector/MessageCollector.java:80]: Could we take the logging back? Possibly as a debug level

**Review by kyrrek — APPROVED** [review]

**Pull request merged**